test/infrastructure/docker/internal/controllers/dockermachine_controller.go

```diff
@@ -195,6 +195,11 @@ func (r *DockerMachineReconciler) reconcileNormal(ctx context.Context, cluster *
 
 		if externalMachine.Exists() {
 			conditions.MarkTrue(dockerMachine, infrav1.ContainerProvisionedCondition)
+			// Setting machine address is required after move, because status.Address field is not retained during move.
+			if err := setMachineAddress(ctx, dockerMachine, externalMachine); err != nil {
+				log.Error(err, "failed to get the machine address")
+				return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
```

Comment:
Not sure if we should rather return the error which gives us an exponential backoff

Reply:
+1 to exponential backoff + surfacing the error in the logs

Reply:
+1 exponential

Reply:
sure, changed it to return error wrapped with the err msg.